app/org/sagebionetworks/bridge/upload/UploadValidationTask.java

```diff
@@ -0,0 +1,59 @@
+package org.sagebionetworks.bridge.upload;
+
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.sagebionetworks.bridge.dao.UploadDao;
+import org.sagebionetworks.bridge.models.upload.UploadStatus;
+
+public class UploadValidationTask implements Runnable {
+    private static final Logger LOG = LoggerFactory.getLogger(UploadValidationTask.class);
+
+    private final UploadValidationContext context;
+
+    private List<UploadValidationHandler> handlerList;
+    private UploadDao uploadDao;
+
+    /* package-scoped */ UploadValidationTask(UploadValidationContext context) {
+        this.context = context;
+    }
+
+    public void setHandlerList(List<UploadValidationHandler> handlerList) {
+        this.handlerList = handlerList;
+    }
+
+    public void setUploadDao(UploadDao uploadDao) {
+        this.uploadDao = uploadDao;
+    }
+
+    @Override
+    public void run() {
+        for (UploadValidationHandler oneHandler : handlerList) {
+            try {
+                oneHandler.handle(context);
+            } catch (UploadValidationException ex) {
```

Comment:
Do you want this to catch any unintended exceptions as well? If not, how will we know if these background tasks are failing? Just from logs?